ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java

```diff
@@ -84,8 +85,16 @@ public void sql(final String sql, final Map<String, Object> overriddenProperties
     }
   }
 
+  /**
+   * @deprecated use {@link #getPersistentQueries}.
+   */
+  @Deprecated
   public Set<QueryMetadata> getRunningQueries() {
-    return ksqlEngine.getLivePersistentQueries();
+    return new HashSet<>(ksqlEngine.getPersistentQueries());
+  }
+
+  public List<PersistentQueryMetadata> getPersistentQueries() {
```

Comment:
Why the preference for `List` over `Set`?

Reply:
Mainly as the `Set` version is no longer exposed by the engine.

Reply:
Do you mean in general on in this specific instance?

In general there's no one size fits all. It depends on the use. Set is good when you're saying 'here is a, probably _unordered_, collection where each entry is on in there _once_'. Where as List is good when you're saying 'here is a _ordered_ collection, but the same entry may appear in it more than once.

`Set` pays the cost on insert of checking the new item against all existing items to ensure uniqueness - this can be quite an overhead if there are many items. 

For this specific instance, `Set` may actually be more correct as we want the _unique_ collection of live persistent queries, but I don't think it's particularly a big issue.  I went with list as that's what fell out of the refactor. It would of been additional work to _correctly_ implement it as a `Set`.  Uniqueness within the set should be based upon the id of the query, which was not the case of the existing set.  It's worth mentioning this it's not just a case of switching `PerstistentQueryMetadata`'s `hashCode` and `equals` to only look at the queryId.  This is a common _hack_, but not good practice.